micrometer-observation-test/src/main/java/io/micrometer/core/tck/ObservationRegistryCompatibilityKit.java

```diff
@@ -187,12 +191,13 @@ void checkedRunnableThrowingErrorShouldBeObserved() {
 
     @Test
     void supplierShouldBeObserved() {
+        @SuppressWarnings("unchecked")
         ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
         when(handler.supportsContext(isA(Observation.Context.class))).thenReturn(true);
         registry.observationConfig().observationHandler(handler);
         Observation observation = Observation.createNotStarted("myObservation", registry);
 
-        String result = observation.observe((Supplier<String>) () -> {
```

Comment:
I did these for two reasons:
1. It is explicit what is tested
2. It is safer since in some cases the cast is needed, this ensures that the right thing is tested even if we change things on the observation or on the tests side

I'm ok removing these though, what do you think?

Reply:
@jonatan-ivanov Thanks for the feedback!

I see your point. I just changed it as IDE suggested. How about extracting it to a local variable to make it explicit as you mentioned while IDE won't suggest?

Reply:
That's a great idea! :)

Reply:
@jonatan-ivanov Thanks! I updated as discussed.

Reply:
👍🏼 Thank you!